### PR TITLE
feat: Add ethereum account abstraction EIP 4337

### PR DESCRIPTION
**Pull request by datanexus-vincent**
Adding these methods as an addon: https://eips.ethereum.org/EIPS/eip-4337#rpc-methods-eth-namespace

I'm still working on confirming that the verification works correctly.

provider.yml
```
endpoints:
    - api-interface: jsonrpc
      chain-id: ETH1
      network-address:
        address: "127.0.0.1:2220"
      node-urls: 
        - url: https://eth-mainnet.g.alchemy.com/v2/{API_KEY}
          addons:
            - bundler
    - api-interface: jsonrpc
      chain-id: SEP1
      network-address:
        address: "127.0.0.1:2220"
      node-urls: 
        - url: https://eth-sepolia.g.alchemy.com/v2/{API_KEY}
          addons:
            - bundler
```
Line 101 of test_spec_full.sh
```
lavad tx pairing stake-provider "$index" $PROVIDERSTAKE "$PROVIDER1_LISTENER,1,bundler" 1 $(operator_address) -y --from servicer1 --delegate-limit 50000000000000ulava --provider-moniker "provider-$index" --gas-adjustment "1.5" --gas "auto" --gas-prices $GASPRICE
```

### CODE REVIEW

File: cookbook/specs/spec_add_ethereum.json [hunk 1]
```diff
@@ -1353,6 +1353,129 @@
...
+                                "category": {
+                                    "deterministic": false,
+                                    "local": true,
+                                    "subscription": false,
+                                    "stateful": 0
+                                },
+                                "extra_compute_units": 0
+                            },
+                            {
+                                "name": "eth_supportedEntryPoint",
```
Comment (omerlavanet) [inline cookbook/specs/spec_add_ethereum.json:1439]: type in the api name, caused the verification to fail
```suggestion
                                "name": "eth_supportedEntryPoints",
```

**Review by omerlavanet — CHANGES_REQUESTED** [review]

File: cookbook/specs/spec_add_ethereum.json (hunk 1 above)
Comment (datanexus-vincent) [reply cookbook/specs/spec_add_ethereum.json]: Can't believe I missed that, sorry. Thank you for tracking it down!

**Comment by datanexus-vincent** [comment]
Verification works as intended after fixing the typo. Bundler endpoints tested with Alchemy, which supports them.

File: cookbook/specs/spec_add_ethereum.json [hunk 2]
```diff
@@ -1353,6 +1353,129 @@
...
+                    },
+                    {
+                        "enabled": true,
+                        "collection_data": {
+                            "api_interface": "jsonrpc",
+                            "internal_path": "",
+                            "type": "POST",
+                            "add_on": "bundler"
+                        },
+                        "apis": [
```
Comment (oren-lava) [inline cookbook/specs/spec_add_ethereum.json:1365]: I noticed that all the APIs are defined with `"deterministic": false`. Why? some of them are deterministic

**Review by oren-lava — CHANGES_REQUESTED** [review]

File: cookbook/specs/spec_add_ethereum.json (hunk 2 above)
Comment (datanexus-vincent) [reply cookbook/specs/spec_add_ethereum.json:1365]: Sorry, not sure how I missed that. I'm fairly certain the `eth_sendUserOperation` and `eth_supportedEntryPoints` are non-deterministic, but the rest are definitely deterministic.

**Review by omerlavanet — APPROVED** [review]

**Review by oren-lava — APPROVED** [review]

**Pull request merged**